hypothesis-python/tests/conjecture/test_provider.py

```diff
@@ -117,18 +114,28 @@ def draw_float(
             return -math.inf
 
         # get rid of infs, they cause nans if we pass them to prng.uniform
+        min_value_bound = min_value
+        max_value_bound = max_value
         if min_value in [-math.inf, math.inf]:
-            min_value = math.copysign(1, min_value) * sys.float_info.max
+            min_value_bound = math.copysign(1, min_value) * sys.float_info.max
             # being too close to the bounds causes prng.uniform to only return
             # inf.
-            min_value /= 2
+            min_value_bound /= 2
         if max_value in [-math.inf, math.inf]:
-            max_value = math.copysign(1, max_value) * sys.float_info.max
-            max_value /= 2
-
-        value = self.prng.uniform(min_value, max_value)
+            max_value_bound = math.copysign(1, max_value) * sys.float_info.max
+            max_value_bound /= 2
+
+        value = self.prng.uniform(min_value_bound, max_value_bound)
+        # random.uniform can in fact provide out of bounds values, e.g.
+        #   random.uniform(-8.98846567431158e+307, 8.98846567431158e+307)
+        # can produce math.inf, which is strictly greater than 8.98846567431158e+307
+        value = clamp(min_value, value, max_value)
```

Comment:
I think that this is a bug we should report upstream to CPython, since they explicitly document that `a <= N <= b`.  I think the balance of performance vs precision favors clamping upstream rather than rejection sampling (it _definitely_ does here).  Plus... we could sneak in another property-based test 😉 

Do you want to send them a PR, or shall I?

Reply:
This might be from our artificial `st.random` instance now that I think of it, I'll double check and reproduce this first

Reply:
yup, it really is a bug:

```
>>> import math
>>> min_value = -8.98846567431158e+307
>>> max_value = 8.98846567431158e+307
>>> import random
>>> assert min_value <= random.uniform(min_value, max_value) <= max_value
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AssertionError
>>> 
```

I'll send them a PR, I should get familiar with how cpython stubs out hypothesis anyway